testcases/OpTestKexec.py

```diff
@@ -0,0 +1,118 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+# IBM_PROLOG_END_TAG
+#
+
+import os
+import pexpect
+import unittest
+import time
+import tempfile
+
+import OpTestConfiguration
+from common import OpTestInstallUtil
+from common.OpTestSystem import OpSystemState
+from common.Exceptions import CommandFailed
+from common.OpTestConstants import OpTestConstants as BMC_CONST
+from common.OpTestError import OpTestError
+
+
+class OpTestKexec(unittest.TestCase):
+
+    def setUp(self):
+        conf = OpTestConfiguration.conf
+        self.cv_SYSTEM = conf.system()
+        self.cv_HOST = conf.host()
+        self.cv_BMC = conf.bmc()
+        self.bmc_type = conf.args.bmc_type
+        self.util = self.cv_SYSTEM.util
+        self.pdbg = conf.args.pdbg
+        self.basedir = conf.basedir
+        self.c = self.cv_SYSTEM.console
+        self.kernel_image = conf.args.kernel_image
+        self.initrd_image = conf.args.initrd_image
+        if self.bmc_type == "FSP_PHYP":
+            self.is_lpar = True
+            self.hmc_user = conf.args.hmc_username
+            self.hmc_password = conf.args.hmc_password
+            self.hmc_ip = conf.args.hmc_ip
+            self.lpar_name = conf.args.lpar_name
+            self.system_name = conf.args.system_name
+            self.cv_HMC = self.cv_SYSTEM.hmc
+            try: self.cpu_resource = conf.args.cpu_resource
+            except AttributeError:
+                self.cpu_resource = 1
+            try: self.mem_resource = conf.args.mem_resource
+            except AttributeError:
+                self.mem_resource = 2048
+        try: self.url = conf.args.url
+        except AttributeError: 
+            self.url = "http://liquidtelecom.dl.sourceforge.net/project/ebizzy/ebizzy/0.3/ebizzy-0.3.tar.gz"
+        self.cv_SYSTEM.goto_state(OpSystemState.OS)
+        res = self.c.run_command("cat /etc/os-release")
+        if "Ubuntu" in res[0] or "Ubuntu" in res[1]:
+            self.distro = "ubuntu"
+        elif 'Red Hat' in res[0] or 'Red Hat' in res[1]:
+            self.distro = 'rhel'
+        elif 'SLES' in res[0] or 'SLES' in res[1]:
+            self.distro = 'sles'
+        else:
+            raise self.skipTest("Test currently supported only on ubuntu, sles and rhel")
+
+    def setup_test(self, dump_place="local"):
+        if dump_place == "local":
+            self.crash_content = self.c.run_command(
+                "ls -l /var/crash | grep '^d'| awk '{print $9}'")
+
+    ##
+    # @brief This function will test the kexec version
+    #
+    def kexec_version(self):
+        self.c.run_command("uname -a")
+        self.c.run_command("cat /etc/os-release")
+        my_cmd = "kexec --version"
+        self.c.run_command(my_cmd)
```

Comment:
I don't think we need this function. Kexec version print in only for informational purpose which can be printed in runTest.